unit_test/batched/Test_Batched_SerialGemm.hpp

```diff
@@ -187,3 +277,65 @@ int test_batched_gemm() {
   
   return 0;
 }
+
+template<typename DeviceType, 
+         typename ValueType, 
+         typename ScalarType,
+         typename ParamTagType,
+         typename AlgoTagType>
+int test_batched_gemm_half() {
```

Comment:
again why the name if nothing actually requires this to be half precision? I mean you could instantiate this with double precision views or?

Reply:
same as https://github.com/kokkos/kokkos-kernels/pull/794#discussion_r496779387. you could instantiate this with any scalar type.

Reply:
Then the name shouldn't have half, right ?

Reply:
I will remove this function and add branches via `std::is_same<ScalarType,half_t` to `impl_test_batched_gemm_half`. @srajama1: Is there anything else that you'd like addressed?